api/admin/controller/library_settings.py

```diff
@@ -294,19 +297,42 @@ def validate_geographic_areas(self, values):
                     elif len([x for x in us_search.query(state=state, returns=None) if x.county == city_or_county]):
                         locations["US"].append(value);
                     else:
-                        return UNKNOWN_LOCATION.detailed(_('Unable to locate "%(value)s".', value=value))
-
+                        if self.ask_registry(value):
+                            return self.ask_registry(value)
                 elif value.isdigit():
                     # Is it a US zipcode?
                     info = us_search.by_zipcode(value)
                     if not info:
                         return UNKNOWN_LOCATION.detailed(_('"%(value)s" is not a valid U.S. zipcode.', value=value))
                     locations["US"].append(self.format_place(value, info.major_city, info.state));
                 else:
-                    return UNKNOWN_LOCATION.detailed(_('Unable to locate "%(value)s".', value=value))
+                    if self.ask_registry(value):
+                        return self.ask_registry(value)
```

Comment:
`ask_registry` gives either a problem detail or `None`, and you'll want to do something other than return if the return value is `None`. I'm not sure exactly how you'll end up there, but if `ask_registry` says everything is okay, you'll eventually want to end up down at `return json.dumps(locations)`.